docs/tracing.md

```diff
@@ -0,0 +1,143 @@
+
+In this document, we document the quirks of our custom support for the `OTEL_TRACES_EXPORTER`,
+as well as examples on how to use tracing, create traceable headers, and how
+to use the Jaeger UI. The [Gateway examples](../examples/gateway/) fully support Tracing.
+
+- [Environment Variables](#environment-variables)
+  - [`OTEL_TRACES_EXPORTER`](#otel_traces_exporter)
+  - [`OTLP Exporter`](#otlp-exporter)
+  - [`Jaeger Exporter`](#jaeger-exporter)
+  - [`Zipkin Exporter`](#zipkin-exporter)
+  - [`File Exporter`](#file-exporter)
+  - [`OTEL_PROPAGATORS`](#otel_propagators)
+- [Using Jaeger UI](#using-jaeger-ui)
+- [Generate `traceparent` Header](#generate-traceparent-header)
+
+## Environment Variables
+
+For advanced configurations, such as ratio-based sampling, please see also the
+[OpenTelemetry Environment Variable Specification].
+
+### `OTEL_TRACES_EXPORTER`
+
+Specifies the exporters to use as a comma-separated string. Each exporter has a
+set of additional environment variables used to configure it. The following values
+are supported:
+
+- `otlp`
+- `jaeger`
+- `zipkin`
+- `stdout`
+- `file` -- appends traces to a JSON file on the filesystem
+
+Default: `""` (no exporters)
+
+### `OTLP Exporter`
+
+Unless specified in this section, the OTLP exporter uses the environment variables
+documented in [OpenTelemetry Protocol Exporter].
+
+#### `OTEL_EXPORTER_OTLP_PROTOCOL`
+Specifies the OTLP protocol to use, which is one of:
+
+- `grpc`
+- `http/protobuf`
+
+Default: `"grpc"`
+
+### `Jaeger Exporter`
+
+See [Jaeger Exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#jaeger-exporter).
+
+### `Zipkin Exporter`
+
+See [Zipkin Exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#zipkin-exporter).
+
+### `File Exporter`
+
+#### `OTEL_EXPORTER_FILE_PATH`
+
+Specifies the filesystem path for the JSON file.
+
+Default: `"$PWD/traces.json"`
+
+### `OTEL_PROPAGATORS`
+
+See [General SDK Configuration](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration).
+
+## Using Jaeger UI
+
+One can use the `jaegertracing/all-in-one` Docker image to run a full Jaeger stack
+and configure the gateway examples, or Kubo, to publish traces to it. Here, in an
+ephemeral container:
+
+```console
+$ docker run --rm -it --name jaeger \
+    -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
+    -p 5775:5775/udp \
+    -p 6831:6831/udp \
+    -p 6832:6832/udp \
+    -p 5778:5778 \
+    -p 16686:16686 \
+    -p 14268:14268 \
+    -p 14269:14269 \
+    -p 14250:14250 \
+    -p 9411:9411 \
+    jaegertracing/all-in-one
+```
+
+Then, in other terminal, start the CAR Gateway example (or Kubo), with Jaeger
+tracing enabled:
+
+```
+$ OTEL_TRACES_EXPORTER=jaeger ./car-gateway -c data.car -p 8040
+```
```

Comment:
a copy-paste example for Kubo would be nice, as that's a common use case here (probably more common that CAR gateway?)